pkg/image/registry/imagesignature/rest.go

```diff
@@ -0,0 +1,93 @@
+package imagesignature
+
+import (
+	"errors"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kapierrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/rest"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	"github.com/openshift/origin/pkg/image/registry/image"
+)
+
+// REST implements the RESTStorage interface for ImageSignature
+type REST struct {
+	imageRegistry image.Registry
+}
+
+// NewREST returns a new REST.
+func NewREST(imageRegistry image.Registry) *REST {
+	return &REST{imageRegistry: imageRegistry}
+}
+
+// New is only implemented to make REST implement RESTStorage
+func (r *REST) New() runtime.Object {
+	return &imageapi.ImageSignature{}
+}
+
+func (r *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, error) {
+	signature := obj.(*imageapi.ImageSignature)
+
+	if err := rest.BeforeCreate(Strategy, ctx, obj); err != nil {
+		return nil, err
+	}
+
+	imageName, _, err := imageapi.SplitImageSignatureName(signature.Name)
+	if err != nil {
+		return nil, kapierrors.NewBadRequest("ImageSignature name must be composed of <imageName>#<signatureName>")
+	}
+
+	image, err := r.imageRegistry.GetImage(ctx, imageName)
+	if err != nil {
+		return nil, err
+	}
+
+	// ensure that given signature already doesn't exist - either by its name or type:content
+	if byName, byContent := imageapi.IndexOfImageSignatureByName(image.Signatures, signature.Name), imageapi.IndexOfImageSignature(image.Signatures, signature.Type, signature.Content); byName >= 0 || byContent >= 0 {
+		return nil, kapierrors.NewAlreadyExists(imageapi.Resource("imageSignatures"), signature.Name)
+	}
+
```

Comment:
This looks like the spot to perform a check of the signature.  We could do it here before doing it inside of a controller.